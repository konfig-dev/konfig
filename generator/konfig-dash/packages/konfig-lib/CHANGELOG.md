# konfig-lib

## 1.36.0

### Minor Changes

- [#450](https://github.com/konfig-dev/konfig/pull/450) [`c002b6f3`](https://github.com/konfig-dev/konfig/commit/c002b6f337801ea3eb71acfef6d49c9f2d8a84d0) Thanks [@dphuang2](https://github.com/dphuang2)! - Handle nullable schema using allOf and $ref

## 1.35.0

### Minor Changes

- [#435](https://github.com/konfig-dev/konfig/pull/435) [`dcbb96d0`](https://github.com/konfig-dev/konfig/commit/dcbb96d0886c811f7271c4a8db3c69f9f6e1c833) Thanks [@dphuang2](https://github.com/dphuang2)! - Adds useAxios0_27_2 to TypeScript SDK configuration

## 1.34.3

### Patch Changes

- [#431](https://github.com/konfig-dev/konfig/pull/431) [`5de1be39`](https://github.com/konfig-dev/konfig/commit/5de1be39ded180f26cec84b36cbf2860d66d07b5) Thanks [@dphuang2](https://github.com/dphuang2)! - fix generation of STATISTICS.md hanging

## 1.34.2

### Patch Changes

- [#420](https://github.com/konfig-dev/konfig/pull/420) [`6e27b58d`](https://github.com/konfig-dev/konfig/commit/6e27b58d3bb3cab5c9616ebbe2275d85e9bf0cb4) Thanks [@eddiechayes](https://github.com/eddiechayes)! - add maptags to transformspec

## 1.34.1

### Patch Changes

- [#404](https://github.com/konfig-dev/konfig/pull/404) [`816adff9`](https://github.com/konfig-dev/konfig/commit/816adff9500ccb56ba35c68152a02b3e8ec022d7) Thanks [@eddiechayes](https://github.com/eddiechayes)! - Remove empty defaults in transformSpec for go generator

## 1.34.0

### Minor Changes

- [#399](https://github.com/konfig-dev/konfig/pull/399) [`7f82ce28`](https://github.com/konfig-dev/konfig/commit/7f82ce28482db4bacabf20eb7283c4f4678729bb) Thanks [@dphuang2](https://github.com/dphuang2)! - add the openapi-schema-validate utility command

- [#399](https://github.com/konfig-dev/konfig/pull/399) [`7f82ce28`](https://github.com/konfig-dev/konfig/commit/7f82ce28482db4bacabf20eb7283c4f4678729bb) Thanks [@dphuang2](https://github.com/dphuang2)! - allow running "konfig lint" without "konfig.yaml" in CWD

## 1.33.0

### Minor Changes

- [#397](https://github.com/konfig-dev/konfig/pull/397) [`1e7398dc`](https://github.com/konfig-dev/konfig/commit/1e7398dcc55c23211b77937bbcfb6746a1c2dcae) Thanks [@dphuang2](https://github.com/dphuang2)! - add mockServerPort to generator configuaration for generated tests

## 1.32.0

### Minor Changes

- [#388](https://github.com/konfig-dev/konfig/pull/388) [`7ca27279`](https://github.com/konfig-dev/konfig/commit/7ca27279a2b13ff88812e5dc16cd9730680b2266) Thanks [@dphuang2](https://github.com/dphuang2)! - transform spec for Java to handle anyOf of only object type schemas

## 1.31.0

### Minor Changes

- [#375](https://github.com/konfig-dev/konfig/pull/375) [`d0f443c5`](https://github.com/konfig-dev/konfig/commit/d0f443c598a2d7c4ded710eb1c1d786aea20b37a) Thanks [@dphuang2](https://github.com/dphuang2)! - convert anyOf schemas to any for Java SDKs

- [#379](https://github.com/konfig-dev/konfig/pull/379) [`234f77a7`](https://github.com/konfig-dev/konfig/commit/234f77a75464ba458edafab91624153b60984577) Thanks [@dphuang2](https://github.com/dphuang2)! - add defaultAsyncTimeout configuration to Python SDK

## 1.30.2

### Patch Changes

- [`cc7c73f2`](https://github.com/konfig-dev/konfig/commit/cc7c73f2590ac4827eef94754482dc58b83e0960) Thanks [@eddiechayes](https://github.com/eddiechayes)! - apply filterPaths to konfig fix

## 1.30.1

### Patch Changes

- [`e5213a41`](https://github.com/konfig-dev/konfig/commit/e5213a41cbea37716363345da920e3612d05ed4c) Thanks [@eddiechayes](https://github.com/eddiechayes)! - add doc to typescriptconfig pagination

- [#354](https://github.com/konfig-dev/konfig/pull/354) [`c57aded2`](https://github.com/konfig-dev/konfig/commit/c57aded2618c791fd7442a01b9e6533b2a394450) Thanks [@eddiechayes](https://github.com/eddiechayes)! - maven test run in quiet mode

- [`4522afbc`](https://github.com/konfig-dev/konfig/commit/4522afbc9b3e44b52fa73e55457de2992742a7ed) Thanks [@eddiechayes](https://github.com/eddiechayes)! - fix defaultchangesetbumptype bug

## 1.30.0

### Minor Changes

- [#355](https://github.com/konfig-dev/konfig/pull/355) [`107261c4`](https://github.com/konfig-dev/konfig/commit/107261c4821f9165d012b8f85b0b8248bae9acd2) Thanks [@dphuang2](https://github.com/dphuang2)! - add googleAnalyticsId configuration for API portal

## 1.29.0

### Minor Changes

- [#351](https://github.com/konfig-dev/konfig/pull/351) [`e19a56d9`](https://github.com/konfig-dev/konfig/commit/e19a56d9cd7337a30070ba56f78083f61a27da92) Thanks [@dphuang2](https://github.com/dphuang2)! - allow default changeset bump type to be configured for a generator

## 1.28.0

### Minor Changes

- [#343](https://github.com/konfig-dev/konfig/pull/343) [`c88285d0`](https://github.com/konfig-dev/konfig/commit/c88285d03321f2d1fedffdfa6164bc22cba92f2b) Thanks [@dphuang2](https://github.com/dphuang2)! - add readmeHeaderSnippet configuration

## 1.27.0

### Minor Changes

- [#323](https://github.com/konfig-dev/konfig/pull/323) [`962ab778`](https://github.com/konfig-dev/konfig/commit/962ab7789efa4af28efed53ecf38b33af1a8800a) Thanks [@dphuang2](https://github.com/dphuang2)! - update description of pythonResponseType

## 1.26.1

### Patch Changes

- [#314](https://github.com/konfig-dev/konfig/pull/314) [`6e163cc8`](https://github.com/konfig-dev/konfig/commit/6e163cc87d32aea6bb3e62ddd1b8e1c3e7b00690) Thanks [@eddiechayes](https://github.com/eddiechayes)! - Fix kill port in containers

## 1.26.0

### Minor Changes

- [#310](https://github.com/konfig-dev/konfig/pull/310) [`ce10a4a1`](https://github.com/konfig-dev/konfig/commit/ce10a4a196734dacc2dee760e9f98501b755de01) Thanks [@dphuang2](https://github.com/dphuang2)! - add hideNonSdkSnippets configuration to API portal

## 1.25.1

### Patch Changes

- [#295](https://github.com/konfig-dev/konfig/pull/295) [`1735540b`](https://github.com/konfig-dev/konfig/commit/1735540b83b9ed7316fd73f9523ba679873c504e) Thanks [@eddiechayes](https://github.com/eddiechayes)! - added filterPaths to konfig.yaml

## 1.25.0

### Minor Changes

- [#291](https://github.com/konfig-dev/konfig/pull/291) [`c90dc42f`](https://github.com/konfig-dev/konfig/commit/c90dc42f1c9e32df66a1390ce532ddecf90c0eb2) Thanks [@dphuang2](https://github.com/dphuang2)! - support light and dark mode in api portal

## 1.24.0

### Minor Changes

- [#271](https://github.com/konfig-dev/konfig/pull/271) [`128b79c1`](https://github.com/konfig-dev/konfig/commit/128b79c1770661b787a17c17571ad447cd134567) Thanks [@dphuang2](https://github.com/dphuang2)! - support globally required security in TypeScript SDK

## 1.23.0

### Minor Changes

- [#263](https://github.com/konfig-dev/konfig/pull/263) [`99310fc1`](https://github.com/konfig-dev/konfig/commit/99310fc166af9212c69aea6a8447bd421f059575) Thanks [@dphuang2](https://github.com/dphuang2)! - add "order" property to konfig.yaml for ordering operations in API reference page

## 1.22.0

### Minor Changes

- [#260](https://github.com/konfig-dev/konfig/pull/260) [`93d5dcf9`](https://github.com/konfig-dev/konfig/commit/93d5dcf94d62ce83549ae7e6985e93713731fce0) Thanks [@dphuang2](https://github.com/dphuang2)! - strip format: uuid for C# SDK

## 1.21.0

### Minor Changes

- [#258](https://github.com/konfig-dev/konfig/pull/258) [`ba01dc5f`](https://github.com/konfig-dev/konfig/commit/ba01dc5f8354bb09dfcbe97917a6c00368128a4b) Thanks [@dphuang2](https://github.com/dphuang2)! - add readmeHeader configuration for more welcoming README pages

## 1.20.0

### Minor Changes

- [#254](https://github.com/konfig-dev/konfig/pull/254) [`d7ebabf8`](https://github.com/konfig-dev/konfig/commit/d7ebabf8f9f528eda5586ae994b9676ff5f711b8) Thanks [@dphuang2](https://github.com/dphuang2)! - support hideOperations for API portal

## 1.19.0

### Minor Changes

- [#250](https://github.com/konfig-dev/konfig/pull/250) [`32dfd9a1`](https://github.com/konfig-dev/konfig/commit/32dfd9a15aaf9016624d173977bd43e3f0725d1d) Thanks [@dphuang2](https://github.com/dphuang2)! - make setSkipSerializationToTrueByDefault true by default

## 1.18.0

### Minor Changes

- [#230](https://github.com/konfig-dev/konfig/pull/230) [`82553c73`](https://github.com/konfig-dev/konfig/commit/82553c73fc836af4c7ea220fbecf375d2fa4b14b) Thanks [@dphuang2](https://github.com/dphuang2)! - add label to link group in documentation configuration

## 1.17.0

### Minor Changes

- [#232](https://github.com/konfig-dev/konfig/pull/232) [`6c28822e`](https://github.com/konfig-dev/konfig/commit/6c28822eb8ccfa53359d7ba105340ffa9409fcad) Thanks [@dphuang2](https://github.com/dphuang2)! - fix "example" property being applied to the wrong places by transformSpec.ts

## 1.16.0

### Minor Changes

- [#228](https://github.com/konfig-dev/konfig/pull/228) [`ab7e69b6`](https://github.com/konfig-dev/konfig/commit/ab7e69b68ea891c74c7fa2c042beb6cd644382ee) Thanks [@dphuang2](https://github.com/dphuang2)! - if schema includes default value and no example then populate the example with the default value

## 1.15.0

### Minor Changes

- [#226](https://github.com/konfig-dev/konfig/pull/226) [`3bfa15ce`](https://github.com/konfig-dev/konfig/commit/3bfa15ce74f365b7b248739c3d84d07458cfdbb3) Thanks [@dphuang2](https://github.com/dphuang2)! - add support for custom dependencies in TypeScript SDK

### Patch Changes

- [#222](https://github.com/konfig-dev/konfig/pull/222) [`d90d63ea`](https://github.com/konfig-dev/konfig/commit/d90d63ea5c472a42be5a0d3c649ad62c955b9a9c) Thanks [@eddiechayes](https://github.com/eddiechayes)! - add pr-create CLI command and /prCreate endpoint

## 1.14.0

### Minor Changes

- [#211](https://github.com/konfig-dev/konfig/pull/211) [`95538406`](https://github.com/konfig-dev/konfig/commit/95538406638712c2b32a323a74742a21da1d5bf0) Thanks [@dphuang2](https://github.com/dphuang2)! - invalid-required-property-syntax lint rule

## 1.13.0

### Minor Changes

- [#206](https://github.com/konfig-dev/konfig/pull/206) [`0b6c29d2`](https://github.com/konfig-dev/konfig/commit/0b6c29d2aa339e2f8755030d03191c21f9affd55) Thanks [@konfig-publisher](https://github.com/konfig-publisher)! - add dart sdk support in konfig.yaml

## 1.12.0

### Minor Changes

- [`97752b65`](https://github.com/konfig-dev/konfig/commit/97752b650be5f5089d8ade979516f9ddc008cd33) Thanks [@dphuang2](https://github.com/dphuang2)! - add initial configurations for dart

## 1.11.0

### Minor Changes

- [#204](https://github.com/konfig-dev/konfig/pull/204) [`32c2e0f4`](https://github.com/konfig-dev/konfig/commit/32c2e0f40c90b1a6caab0d2f3c6a0c098de0608a) Thanks [@dphuang2](https://github.com/dphuang2)! - added required property to client state

## 1.10.0

### Minor Changes

- [#198](https://github.com/konfig-dev/konfig/pull/198) [`2b90fa13`](https://github.com/konfig-dev/konfig/commit/2b90fa132bceb427272d9835686977a51eda01f1) Thanks [@dphuang2](https://github.com/dphuang2)! - add favicon/logo configuration for portal

## 1.9.0

### Minor Changes

- [#190](https://github.com/konfig-dev/konfig/pull/190) [`5d046896`](https://github.com/konfig-dev/konfig/commit/5d04689669cced6f4462e7b24a8b7462ad00798c) Thanks [@dphuang2](https://github.com/dphuang2)! - label is optional for markdown documentation

## 1.8.1

### Patch Changes

- [#188](https://github.com/konfig-dev/konfig/pull/188) [`fab62fdf`](https://github.com/konfig-dev/konfig/commit/fab62fdf7f5b0ca3369f7f326dc9a8a4d56b77ed) Thanks [@dphuang2](https://github.com/dphuang2)! - modify doc config to reuse linkSchema

## 1.8.0

### Minor Changes

- [#177](https://github.com/konfig-dev/konfig/pull/177) [`28f0b957`](https://github.com/konfig-dev/konfig/commit/28f0b957d4108d1eec8c129da718a56bd594672c) Thanks [@dphuang2](https://github.com/dphuang2)! - add documentation configuration to konfig.yaml

## 1.7.0

### Minor Changes

- [#174](https://github.com/konfig-dev/konfig/pull/174) [`3756851e`](https://github.com/konfig-dev/konfig/commit/3756851e22a6adf25ad9612e71118a46ffc8af1d) Thanks [@dphuang2](https://github.com/dphuang2)! - fixInheritMetadataFromSpec

## 1.6.0

### Minor Changes

- [#155](https://github.com/konfig-dev/konfig/pull/155) [`b4366328`](https://github.com/konfig-dev/konfig/commit/b4366328b6c17a95b3d56a200e5d1808593c8528) Thanks [@dphuang2](https://github.com/dphuang2)! - add readmeOperatin.operationId

## 1.5.0

### Minor Changes

- [#143](https://github.com/konfig-dev/konfig/pull/143) [`79cd79f0`](https://github.com/konfig-dev/konfig/commit/79cd79f0380e8c414d0d97dc9e9c97eeed521f2c) Thanks [@dphuang2](https://github.com/dphuang2)! - add portal configuration to top-level konfig.yaml

### Patch Changes

- [#143](https://github.com/konfig-dev/konfig/pull/143) [`79cd79f0`](https://github.com/konfig-dev/konfig/commit/79cd79f0380e8c414d0d97dc9e9c97eeed521f2c) Thanks [@dphuang2](https://github.com/dphuang2)! - remove strictness from KonfigYaml schema for forward compatability

## 1.4.0

### Minor Changes

- [#132](https://github.com/konfig-dev/konfig/pull/132) [`1059bd82`](https://github.com/konfig-dev/konfig/commit/1059bd82a1dde5695d361a4ce56be89364babb29) Thanks [@dphuang2](https://github.com/dphuang2)! - add primaryColor / portalTitle to konfig.yaml

## 1.3.3

### Patch Changes

- [#130](https://github.com/konfig-dev/konfig/pull/130) [`b0010205`](https://github.com/konfig-dev/konfig/commit/b001020545daad367d634c1a290722894976c95a) Thanks [@dphuang2](https://github.com/dphuang2)! - add undeclared dependencies

## 1.3.2

### Patch Changes

- [#128](https://github.com/konfig-dev/konfig/pull/128) [`297d5e45`](https://github.com/konfig-dev/konfig/commit/297d5e450613478d49bbc92357ea2a7e5f677acc) Thanks [@dphuang2](https://github.com/dphuang2)! - make openapi-types a non-dev dependency for konfig-lib

## 1.3.1

### Patch Changes

- [#123](https://github.com/konfig-dev/konfig/pull/123) [`f9ff241c`](https://github.com/konfig-dev/konfig/commit/f9ff241c03c5bd0645cde46ebaf1d0437a86a301) Thanks [@eddiechayes](https://github.com/eddiechayes)! - add konfig changeset cli command

## 1.3.0

### Minor Changes

- [#119](https://github.com/konfig-dev/konfig/pull/119) [`e41378f0`](https://github.com/konfig-dev/konfig/commit/e41378f0f074e03c0cc512c546fd55c1e55d3411) Thanks [@dphuang2](https://github.com/dphuang2)! - - Run "konfig fix" at beginning of "konfig generate" if "specInputPath" is defined
  - fix "waiton" logic for mock server in "konfig test"
  - add prerelease "-r" flag for "konfig bump"

## 1.2.0

### Minor Changes

- [#113](https://github.com/konfig-dev/konfig/pull/113) [`29864036`](https://github.com/konfig-dev/konfig/commit/29864036e04a6e2cdc393e83ca189b7d09bc8323) Thanks [@dphuang2](https://github.com/dphuang2)! - add description to gemName

## 1.1.6

### Patch Changes

- 7f3a6349: add description for filterModels field

## 1.1.5

### Patch Changes

- bb4c6c7d: add description to useSecurityKeyParamNameAsPropertyName

## 1.1.4

### Patch Changes

- c833b8d1: add description for npmName

## 1.1.3

### Patch Changes

- 541aded4: add description to artifactId of Java SDK config

## 1.1.2

### Patch Changes

- 72de0f1: add description to clientName in typescript SDK configuration

## 1.1.0

### Minor Changes

- useDescriptionInOperationTableDocumentation
