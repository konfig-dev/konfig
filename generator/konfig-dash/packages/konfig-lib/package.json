{
  "name": "konfig-lib",
  "packageManager": "yarn@3.2.1",
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "files": [
    "dist/**/*",
    "konfig-yaml.schema.json"
  ],
  "scripts": {
    "dev": "tsc --preserveWatchOutput --watch --pretty",
    "build": "yarn clean && tsc && yarn generate",
    "clean": "rm -rf dist/",
    "generate": "ts-node generate-json-schema.ts",
    "prepack": "yarn build",
    "changeset:publish": "../../bash-scripts/publish.sh",
    "release": "konfig-release-it",
    "release-ci": "konfig-release-it --ci",
    "update-snapshots": "jest --updateSnapshot",
    "test": "jest"
  },
  "dependencies": {
    "@apidevtools/swagger-parser": "^10.1.0",
    "@stoplight/json-ref-resolver": "^3.1.5",
    "camelcase": "5.3.1",
    "chroma-js": "^2.4.2",
    "deep-equal": "^2.2.1",
    "fs-extra": "^10.1.0",
    "js-yaml": "^4.1.0",
    "json-stable-stringify": "^1.0.2",
    "jsonpath-plus": "^7.2.0",
    "konfig-swagger2openapi": "workspace:*",
    "konfig-zod-to-openapi": "workspace:*",
    "mdast-util-from-markdown": "0.8.5",
    "merge-anything": "^5.1.4",
    "openapi-types": "^12.1.0",
    "yaml": "^2.2.1",
    "zod": "^3.20.2"
  },
  "devDependencies": {
    "@stoplight/spectral-core": "^1.16.0",
    "@types/chroma-js": "^2.4.0",
    "@types/deep-equal": "^1.0.1",
    "@types/jest": "^29.0.1",
    "jest": "^29.0.3",
    "konfig-release-it": "workspace:*",
    "ts-jest": "^29.0.0",
    "ts-node": "^10.9.1",
    "typescript": "^4.7.4",
    "zod-to-json-schema": "^3.20.4"
  },
  "version": "1.37.8"
}
