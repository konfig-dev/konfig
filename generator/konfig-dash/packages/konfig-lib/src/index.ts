export * from './GenerateRequestBody'
export * from './KonfigYaml'
export * from './KonfigYamlCommon'
export * from './JavaGenerateApiRequestBody'
export * from './KonfigYamlFiles'
export * from './TestConfig'
export * from './TemplateFiles'
export * from './util/get-package-version'
export * from './util/set-package-version'
export * from './util/konfig-yaml-name'
export * from './util/get-konfig-lint-workflow-path'
export * from './util/find-redundant-security-requirement-and-parameter'
export * from './util/find-parameter-index-by-name-and-in'
export * from './util/get-security-schemes'
export * from './util/default-or-200-range-status-code-regex'
export * from './util/yaml-dump'
export * from './yaml-language-server-comment'
export * from './util/disallowed-header-names-request'
export * from './util/fix-custom-modifications'
export * from './util/can-schema-be-structured'
export * from './url-regex'
export * from './util/get-security-in'
export * from './util/is-object-type-schema-with-no-properties'
export * from './util/get-security-parameter-name'
export * from './util/konfig-api-test-environment-name'
export * from './util/ruleset-json-paths'
export * from './util/is-schema-example-possible-incorrect-data-type'
export * from './util/get-ignore'
export * from './util/resolve-ref-stoplight'
export * from './util/find-security-schemes-for-parameters'
export * from './util/operation-id-schema'
export * from './util/find-redundant-security-schemes'
export * from './util/tag-schema'
export * from './util/is-json-string'
export * from './util/json-schema'
export * from './util/update-or-set-components-schema'
export * from './util/update-request-body-schema-inline'
export * from './util/generate-schema-object-from-json'
export * from './util/generate-and-merge-schema-objects'
export * from './util/get-components-ref-name'
export * from './util/get-operation-parameters-deprecated'
export * from './util/fix-missing-global-tags'
export * from './util/get-example-json-from-media-object'
export * from './util/get-example-jsons-from-media-object'
export * from './util/brand'
export * from './parseSpec'
export * from './transformSpec'
export * from './get-operations'
export * from './resolveRef'
export * from './forEachOperation'
export * from './forEachSchema'
export * from './forEachPath'
export * from './forEachReferenceUntyped'
export * from './recurseObject'
export * from './snaptrade-demo'
