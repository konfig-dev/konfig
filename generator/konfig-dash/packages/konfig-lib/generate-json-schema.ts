import zodToJsonSchema from 'zod-to-json-schema'
import { KonfigYaml } from './src/KonfigYaml'
import * as fs from 'fs-extra'
import * as path from 'path'

const jsonSchema = zodToJsonSchema(KonfigYaml, {
  name: 'konfigYaml',
})

const outputPath = path.join(__dirname, 'konfig-yaml.schema.json')
const docOutputPath = path.join(
  path.dirname(path.dirname(path.dirname(__dirname))),
  'konfig-docs',
  'static',
  'konfig-yaml.schema.json'
)
if (fs.existsSync(outputPath))
  fs.writeFileSync(outputPath, JSON.stringify(jsonSchema, undefined, 2))
if (fs.existsSync(docOutputPath))
  fs.writeFileSync(docOutputPath, JSON.stringify(jsonSchema, undefined, 2))
