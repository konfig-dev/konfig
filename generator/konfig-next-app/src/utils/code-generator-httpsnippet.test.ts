import { HttpMethodsEnum } from 'konfig-lib'
import {
  CodeGeneratorHttpSnippetConstructorArgs,
  CodeGeneratorHttpsnippet,
} from './code-generator-httpsnippet'
import { test, expect } from 'vitest'

const gitConfig = {
  owner: '',
  path: '',
}

test('httpsnippet - HAR validation error', async () => {
  const args: CodeGeneratorHttpSnippetConstructorArgs = {
    contentType: null,
    httpMethod: HttpMethodsEnum.GET,
    path: '/v1/runs/{workflow_run_id}',
    parameters: [
      {
        description: 'The ID of the workflow run to retrieve.',
        name: 'workflow_run_id',
        required: true,
        in: 'path',
        example: 'rnp_x3p27VQk6MyJfLe',
        schema: { type: 'string' },
      },
    ],
    requestBody: null,
    securitySchemes: {
      api_key: { type: 'apiKey', in: 'header', name: 'X-Api-Key' },
    },
    formData: {
      parameters: { workflow_run_id: 'm ' },
      security: {
        api_key: { type: 'apiKey', in: 'header', key: 'X-Api-Key', value: '' },
      },
      requestBody: '',
    },
    languageConfigurations: {
      typescript: {
        clientName: 'Leap',
        packageName: '@leap-ai/workflows',
        git: gitConfig,
      },
      python: {
        clientName: 'Leap',
        packageName: 'leap_workflows',
        projectName: 'leap-python-sdk',
        git: gitConfig,
      },
    },
    servers: ['https://api.workflows.tryleap.ai'],
    operationId: 'WorkflowRuns_getWorkflowRun',
    tag: 'Workflow Runs',
    basePath: 'https://api.workflows.tryleap.ai',
    oauthTokenUrl: null,
    originalOauthTokenUrl: null,
    requestBodyRequired: false,
    targetId: 'shell',
  }
  const code = await new CodeGeneratorHttpsnippet(args).snippet()
  expect(code).toMatchSnapshot()
})

test('httpsnippet - deeply nested objects with files', async () => {
  const args: CodeGeneratorHttpSnippetConstructorArgs = {
    contentType: 'multipart/form-data',
    httpMethod: HttpMethodsEnum.POST,
    path: '/v1/ingest/documents/local',
    parameters: [],
    requestBody: {
      name: '',
      in: 'body',
      schema: {
        type: 'array',
        items: {
          type: 'object',
          required: ['blob', 'metadata'],
          properties: {
            blob: {
              description: 'The actual file being uploaded.',
              type: 'string',
              format: 'binary',
            },
            metadata: {
              type: 'object',
              required: ['bucketId', 'fileName', 'fileType'],
              properties: {
                bucketId: {
                  type: 'integer',
                  example: 1234,
                },
                fileName: {
                  type: 'string',
                  example: 'my_file.txt',
                },
                fileType: {
                  type: 'string',
                  enum: ['txt', 'docx', 'pptx', 'xlsx', 'pdf', 'png', 'jpg'],
                },
                metadata: {
                  type: 'object',
                  example: {
                    key: 'value',
                  },
                },
                callbackData: {
                  type: 'string',
                  example: 'my_callback_data',
                },
                callbackUrl: {
                  type: 'string',
                  example: 'https://my.callback.url.com',
                },
              },
            },
          },
        },
      },
      isRequestBody: true,
    },
    securitySchemes: {
      ApiKeyAuth: {
        type: 'apiKey',
        in: 'header',
        name: 'X-API-Key',
      },
    },
    formData: {
      parameters: {
        documentId: '',
      },
      security: {
        ApiKeyAuth: {
          type: 'apiKey',
          in: 'header',
          key: 'X-API-Key',
          value: 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX',
        },
      },
      requestBody: [
        {
          blob: new File([], 'document.txt'),
          metadata: {
            bucketId: 6124,
            fileName: 'document.txt',
            fileType: 'txt',
            metadata: '',
            callbackData: '',
            callbackUrl: '',
          },
        },
      ],
    },
    languageConfigurations: {
      typescript: {
        clientName: 'Groundx',
        packageName: 'groundx-typescript-sdk',
        git: gitConfig,
      },
      python: {
        clientName: 'Groundx',
        packageName: 'groundx',
        projectName: 'groundx-python-sdk',
        git: gitConfig,
      },
    },
    servers: ['https://api.groundx.ai/api'],
    operationId: 'Document_uploadLocal',
    tag: 'Documents',
    basePath: 'https://api.groundx.ai/api',
    oauthTokenUrl: null,
    originalOauthTokenUrl: null,
    requestBodyRequired: false,
    targetId: 'shell',
  }
  const code = await new CodeGeneratorHttpsnippet(args).snippet()
  expect(code).toMatchSnapshot()
})

test('generate "requestBody" parameter when non-request body parameter and request body value exist', async () => {
  const code = await new CodeGeneratorHttpsnippet({
    contentType: 'application/json',
    httpMethod: HttpMethodsEnum.PUT,
    path: '/quotes/service_groups/{service_group_id}/locations',
    parameters: [
      {
        description: 'UUID of the service group',
        name: 'service_group_id',
        in: 'path',
        required: true,
        schema: {
          title: 'Service Group Id',
          description: 'UUID of the service group',
          type: 'string',
        },
      },
    ],
    requestBody: {
      name: '',
      in: 'body',
      schema: {
        title: 'Location Ids',
        description:
          'List of location IDs to replace the existing locations on the service group',
        type: 'array',
        items: {
          type: 'string',
        },
      },
      isRequestBody: true,
      required: true,
    },
    securitySchemes: {
      Auth0HTTPBearer: {
        type: 'http',
        scheme: 'bearer',
      },
    },
    formData: {
      parameters: {
        service_group_id: '',
      },
      security: {
        Auth0HTTPBearer: {
          type: 'bearer',
          value: '',
        },
      },
      requestBody: ['123'],
    },
    languageConfigurations: {
      typescript: {
        clientName: 'Bellhop',
        packageName: 'bellhop-partners-typescript',
        git: {
          owner: 'konfig-dev',
          path: 'bellhop-sdks/tree/main/typescript',
        },
      },
      python: {
        clientName: 'Bellhop',
        packageName: 'bellhop',
        projectName: 'bellhop-partners-python',
        git: {
          owner: 'konfig-dev',
          path: 'bellhop-sdks/tree/main/python',
        },
      },
    },
    servers: [
      'https://partners.bellhops.dev/v5',
      'https://partners.bellhop.com/v5',
    ],
    operationId: 'QuoteServiceGroups_changeLocations',
    tag: 'QuoteServiceGroups',
    basePath: 'https://partners.bellhops.dev/v5',
    oauthTokenUrl: null,
    originalOauthTokenUrl: null,
    requestBodyRequired: true,
    targetId: 'shell',
    mode: 'copy',
  }).snippet()
  expect(code).toMatchSnapshot()
})
