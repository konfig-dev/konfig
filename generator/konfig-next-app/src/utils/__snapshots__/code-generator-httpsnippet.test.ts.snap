// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generate "requestBody" parameter when non-request body parameter and request body value exist 1`] = `
"curl --request PUT \\\\
  --url https://partners.bellhops.dev/v5/quotes/service_groups/SERVICE_GROUP_ID/locations \\\\
  --header 'Content-Type: application/json' \\\\
  --data-binary '[
  \\"123\\"
]'"
`;

exports[`httpsnippet - HAR validation error 1`] = `
"curl --request GET \\\\
  --url https://api.workflows.tryleap.ai/v1/runs/m%20"
`;

exports[`httpsnippet - deeply nested objects with files 1`] = `
"curl --request POST \\\\
  --url https://api.groundx.ai/api/v1/ingest/documents/local \\\\
  --header 'Content-Type: multipart/form-data' \\\\
  --header 'X-API-Key: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX' \\\\
  --form blob=@document.txt \\\\
  --form 'metadata={\\"bucketId\\":6124,\\"fileName\\":\\"document.txt\\",\\"fileType\\":\\"txt\\"}'"
`;
